lib/bundler/runtime.rb

```diff
@@ -189,8 +189,16 @@ def clean(dry_run = false)
 
       unless dry_run
         stale_files = stale_gem_bins + stale_gem_files + stale_gemspec_files
-        stale_files.each {|file| FileUtils.rm(file) if File.exist?(file) }
-        stale_git_cache_dirs.each {|dir| FileUtils.rm_rf(dir) if File.exist?(dir) }
+        stale_files.each do |file|
+          SharedHelpers.filesystem_access(File.dirname(file)) do |_p|
```

Comment:
Hm, so I was reading up on what the Read-Write-Exec permissions needed for removing a file are and it seems that oddly enough, Write permissions on the directory that the file resides in are needed. Write or execute permissions on that file itself aren't needed since file removal is essentially modifying the directory and its info. (http://unix.stackexchange.com/questions/48579/why-can-rm-remove-read-only-files)

Hence, for the error we potentially throw from the `filesystem_access` method, we'd want to tell the user they need write access on the directory for the file targeted for deletion.